src/services/formatting/formatting.ts

```diff
@@ -493,9 +480,9 @@ namespace ts.formatting {
                     }
                 },
                 getIndentation: () => indentation,
-                getDelta: () => delta,
+                getDelta: (child) => (delta && SmartIndenter.shouldInheritParentIndentation(node, child)) ? 0 : delta,

```

Comment:
to me condition ` SI.shouldInheritParentIndentation() ? 0 : delta` look more clear and it is doing the same thing.
- if delta is not-null-or-zero then body of `getDelta` boils down to `SI.shouldInheritParentIndentation() ? 0 : delta`
- if delta zero then `getDelta` evaluates to just `delta` which will mean the same